theseus/geometry/functions/so3_function.py

```diff
@@ -0,0 +1,202 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+import theseus
+
+from typing import Optional, List, cast
+
+from .lie_group_function import LieGroupExpMap
+from .utils import check_jacobians_list
+
+
+def name() -> str:
+    return "SO3"
+
+
+def dim() -> int:
+    return 3
+
+
+def check_group_tensor(tensor: torch.Tensor) -> bool:
+    with torch.no_grad():
+        if tensor.ndim != 3 or tensor.shape[1:] != (3, 3):
```

Comment:
Should we allow any batch dims (..., 3, 3) rather than just one so things are more torch like?